cabal-install-solver/src/Distribution/Solver/Types/ProjectConfigPath.hs

```diff
@@ -60,9 +62,13 @@ instance Structured ProjectConfigPath
 -- >>> render . docProjectConfigPath $ ProjectConfigPath $ "D.config" :| ["C.config", "B.config", "A.project"]
 -- "D.config\n  imported by: C.config\n  imported by: B.config\n  imported by: A.project"
 docProjectConfigPath :: ProjectConfigPath -> Doc
-docProjectConfigPath (ProjectConfigPath (p :| [])) = text p
-docProjectConfigPath (ProjectConfigPath (p :| ps)) = vcat $
-    text p : [ text " " <+> text "imported by:" <+> text l | l <- ps ]
+docProjectConfigPath (ProjectConfigPath (p :| [])) = quoteUntrimmed p
+docProjectConfigPath (ProjectConfigPath (p :| ps)) = vcat $ quoteUntrimmed p :
+    [ text " " <+> text "imported by:" <+> quoteUntrimmed l | l <- ps ]
+
+-- | If the path has leading or trailing spaces then show it quoted.
+quoteUntrimmed :: FilePath -> Doc
+quoteUntrimmed s = if trim s /= s then quotes (text s) else text s
```

Comment:
I dount the idea to make spaces explicit with quoting and, at the same time, trim those spaces. My understanding of the need for trimming is that spaces should not be there in the first place. So, my proposal would be to issue a warning instead if we notice `trim s /= s`.

Reply:
I'm now issuing an error for this case.

Reply:
Mm, I feel like error is too harsh. I wonder what others think.

Reply:
I've softened the error down to a warning.